internal/pkg/manifest/validate.go

```diff
@@ -1721,6 +1738,47 @@ func validateDepsForEssentialContainers(deps map[string]containerDependency) err
 	return nil
 }
 
+func validateExposedPorts(opts validateExposedPortsOpts) error {
+	exposedPorts := make(map[uint16]string)
+
+	if opts.mainContainerPort != nil {
+		exposedPorts[aws.Uint16Value(opts.mainContainerPort)] = opts.mainContainerName
+	}
+
+	for name, sidecar := range opts.sidecarConfig {
+		if sidecar.Port != nil {
```

Comment:
instead of nesting all of this logic, it'd be nice to just `continue` in the opposite case:
```suggestion
		if sidecar.Port == nil {
		  continue
		}
```